src/frontend/src/utils/recoveryDevice.ts

```diff
@@ -0,0 +1,17 @@
+import { DeviceData, Purpose } from "../../generated/internet_identity_types";
+
+export type RecoveryDevice = Omit<DeviceData, "alias"> & {
```

Comment:
Do `RecoveryDevice`s never have an alias? Also, there are a few occurrences of `RecoveryDevice & Omit<DeviceData, "alias">` and `RecoveryDevice & DeviceData` here and in other modules. Shouldn't these all just be `RecoveryDevice`?

Reply:
Technically they do have an alias, in particular when interacting with authenticated canister calls (i.e. `add` and `update`) the alias is _required_.

Agreed, `RecoveryDevice & Omit<DeviceData, "alias">` is redundant. Thanks for pointing it out. However, `RecoveryDevice & DeviceData` is not, as this combines the existence of `alias` with the restriction on `purpose`.